aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -147,7 +148,17 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     } catch (RuntimeException persistFailure) {
       LOG.error("Confirming if commit to {} indeed failed to persist, attempting to reconnect and check.",
           fullTableName, persistFailure);
-      commitStatus = checkCommitStatus(newMetadataLocation, metadata);
+
+      if (persistFailure instanceof AwsServiceException) {
```

Comment:
Can this be a separate catch block to simplify the exception handling flow? like 

>   } catch (AwsServiceException serviceException) {
> ...
>   } catch (RuntimeException runtimeException) {
> ...
>